### PR TITLE
General update

### PR DESCRIPTION
**Pull request by adamyala**
* add sdk version to header
* change post class from params to dict param
* update unit tests
* readme updates
* content_field supports optional params

### CODE REVIEW

File: butter_cms/__version__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__version__ = '0.8'
```
Comment (adamyala) [inline butter_cms/__version__.py]: Since the version is now referenced in `setup.py` and the request header, I put it in this file. The ideal way to setup the `__version__` attribute would be in `__init__.py` per https://www.python.org/dev/peps/pep-0396/ but I've also seen plenty of packages put it in `__version__.py`. Doing this per the spec caused some import issues.

Comment (jakelumetta) [reply butter_cms/__version__.py]: Seems fine. Stripe puts there's into a `VERSION` file:  https://github.com/stripe/stripe-python/blob/master/VERSION

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+certifi==2017.7.27.1
```
Comment (jakelumetta) [inline requirements.txt]: Why is this file needed? Dependencies are defined here: https://github.com/ButterCMS/buttercms-python/blob/master/setup.py#L22

File: README.md [hunk 3]
```diff
@@ -167,3 +167,11 @@ client.content_fields.get(['homepage_headline', 'homepage_title'])
 ### Other
```
Comment (jakelumetta) [inline README.md]: Please update this file to be current + accurate based on the changes in this PR

File: butter_cms/page.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from .client import Client
+
+
+class Page(Client):
+    """Page"""
+    def __init__(self, auth_token):
+        Client.__init__(self, auth_token)
+        self.path = 'pages/'
+
+    def list(self, slug, params=None):
```
Comment (jakelumetta) [inline butter_cms/page.py]: Let's rename `slug` to `page_type` since that's what the slug represents.

File: butter_cms/page.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        full_slug = '{}/'.format(slug)
+        return self.api_get(slug=full_slug, params=params)
+
+    def get(self, slug1, slug2, params=None):
```
Comment (jakelumetta) [inline butter_cms/page.py]: Rename `slug1` to `page_type`
Rename `slug2` to `page_slug`

File: butter_cms/post.py [hunk 6]
```diff
@@ -10,10 +10,15 @@ def __init__(self, auth_token):
     def all(self, params=None):
```
Comment (jakelumetta) [inline butter_cms/post.py]: Does this call result in a trailing slash? `v2/posts/?...`

**Comment by jakelumetta** [comment]
Can you update the ContentField call to support both `keys` and optional `params` dict ?

https://github.com/adamyala/buttercms-python/blob/e3ad2557d04ff9f18361645ee9332e6cdaa764a9/butter_cms/content_field.py#L10

File: butter_cms/post.py (hunk 6 above)
Comment (adamyala) [reply butter_cms/post.py]: It should. The path for `posts/` is set on https://github.com/ButterCMS/buttercms-python/blob/master/butter_cms/post.py#L8

**Comment by jakelumetta** [comment]
Looking good. Please update https://github.com/ButterCMS/buttercms-python/blob/master/butter_cms/unit_tests.py to include pages; then we should be good to go!

**Comment by jakelumetta** [comment]
Looking good. Please update https://github.com/ButterCMS/buttercms-python/blob/master/butter_cms/unit_tests.py to include pages; then we should be good to go!
